### PR TITLE
Add ability to MPAS to argo with depth

### PR DESCRIPTION
**Pull request by vanroekel**
this is a first cut at implementing the ability to compare against argo with depth.  The data is from Roemmich-Gilson Argo Climatology (http://sio-argo.ucsd.edu/RG_Climatology.html).  I've moved data to nersc for testing.  I would appreciate any suggestions.

### CODE REVIEW

**Comment by vanroekel** [comment]
here is my test http://portal.nersc.gov/project/acme/lvroekel/argoTestEC60to30/

**Comment by xylar** [comment]
@vanroekel, this is awesome!  I will review it more thoroughly in the next couple of days but it looks good at a glance.

We discussed maybe improving the masking of the ARGO data based on the `bathymetry_mask` field.  Let me know when that's on NERSC and I'll test with my favorite run(s).

**Comment by xylar** [comment]
@vanroekel, I'm hoping to test this out soon but the ARGO observations directory on NERSC has the following permissions:
```
drwxrwx---  2 lvroekel lvroekel  512 Mar  7 14:34 ARGO
```
Can you `chgrp` it?

**Comment by vanroekel** [comment]
@xylar permissions should be set now.  If there is a problem let me know.  

I'm still working on the masking.  I'll upload as soon as I can for testing.

**Comment by xylar** [comment]
Great, worked like a charm.  Let me know when you have updated obs that you're happy with.  If you have scripts you used to pre-process the obs, you can put them in `preprocess_observations/`, particularly if you can also include instructions and/or a URL for downloading the original observations.

**Comment by vanroekel** [comment]
@xylar, new data is on nersc now.  Masking looks good to me.  Here is my test
http://portal.nersc.gov/project/acme/lvroekel/argoTest/ocean/index.html

**Comment by xylar** [comment]
@vanroekel, thanks for finding the conflict with the climatologies.  I would suggest we do a corresponding name change for the SOSE output in this PR.  Is that something you want to do or should I take care of it?

Also, I think there's a bit of a risk of confusion with the mapping files related to the observations.  I'll make some inline comments about what we might want to rename to avoid that.

File: mpas_analysis/ocean/climatology_map_argo.py [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+
+            refTitleLabel = 'Roemmich-Gilson Argo Climatology: Temperature'
+
+            observationsDirectory = build_config_full_path(
+                config, 'oceanObservations', 'argoSubdirectory')
+
+            obsFileName = \
+                '{}/ArgoClimatology_TS.nc'.format(
+                        observationsDirectory)
+            refFieldName = 'theta'
```
Comment (xylar) [inline mpas_analysis/ocean/climatology_map_argo.py]: Could we rename this to 'thetaArgo' so the corresponding mapping file is clearly for ARGO?  So far, there's no conflict but I could imagine one in the future.

**Comment by vanroekel** [comment]
Sure, I’m happy to take care of both points.

File: mpas_analysis/ocean/climatology_map_argo.py [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+
+                    subtask.set_plot_info(
+                        outFileLabel=outFileLabel,
+                        fieldNameInTitle='Temperature',
+                        mpasFieldName=mpasFieldName,
+                        refFieldName=refFieldName,
+                        refTitleLabel=refTitleLabel,
+                        diffTitleLabel=diffTitleLabel,
+                        unitsLabel=r'$^\circ$C',
+                        imageCaption='ARGO_Temperature',
```
Comment (xylar) [inline mpas_analysis/ocean/climatology_map_argo.py]: The image caption should describe the figure you're seeing when you have a single image up in the browser.  Could you make this a bit more descriptive? Something like "Temperature compared with ARGO observations"?

File: mpas_analysis/ocean/climatology_map_argo.py [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+
+            refTitleLabel = 'Roemmich-Gilson Argo Climatology: Salinity'
+
+            observationsDirectory = build_config_full_path(
+                config, 'oceanObservations', 'argoSubdirectory')
+
+            obsFileName = \
+                '{}/ArgoClimatology_TS.nc'.format(
+                        observationsDirectory)
+            refFieldName = 'salinity'
```
Comment (xylar) [inline mpas_analysis/ocean/climatology_map_argo.py]: change to "salinityArgo"?

File: mpas_analysis/ocean/climatology_map_argo.py [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+
+                    subtask.set_plot_info(
+                        outFileLabel=outFileLabel,
+                        fieldNameInTitle='Salinity',
+                        mpasFieldName=mpasFieldName,
+                        refFieldName=refFieldName,
+                        refTitleLabel=refTitleLabel,
+                        diffTitleLabel=diffTitleLabel,
+                        unitsLabel=r'PSU',
+                        imageCaption='ARGO_Salinity',
```
Comment (xylar) [inline mpas_analysis/ocean/climatology_map_argo.py]: Similarly to above, "Salinity compared with ARGO observations"

Comment (xylar) [reply mpas_analysis/ocean/climatology_map_argo.py]: In both cases, you could also put the depth in the caption if you wanted but it's not necessary.

File: mpas_analysis/ocean/climatology_map_argo.py (hunk 2 above)
Comment (vanroekel) [reply mpas_analysis/ocean/climatology_map_argo.py]: sure will do.  A question -- where does the imageCaption appear?  I don't see ARGO_temperature in the figures I generate.

File: mpas_analysis/ocean/climatology_map_argo.py (hunk 3 above)
Comment (vanroekel) [reply mpas_analysis/ocean/climatology_map_argo.py]: done

File: mpas_analysis/ocean/climatology_map_argo.py (hunk 4 above)
Comment (vanroekel) [reply mpas_analysis/ocean/climatology_map_argo.py]: done

File: mpas_analysis/ocean/climatology_map_argo.py (hunk 1 above)
Comment (vanroekel) [reply mpas_analysis/ocean/climatology_map_argo.py]: done

**Comment by vanroekel** [comment]
@xylar I also made corresponding changes to the SOSE task, please let me know if they are correct.

File: mpas_analysis/ocean/climatology_map_argo.py [hunk 5]
```diff
@@ -0,0 +1,461 @@
...
+                        fieldNameInTitle='Temperature',
+                        mpasFieldName=mpasFieldName,
+                        refFieldName=refFieldName,
+                        refTitleLabel=refTitleLabel,
+                        diffTitleLabel=diffTitleLabel,
+                        unitsLabel=r'$^\circ$C',
+                        imageCaption='Model temperature compared with ARGO observations',
+                        galleryGroup='Argo Temperature',
+                        groupSubtitle=None,
+                        groupLink='temp',
```
Comment (xylar) [inline mpas_analysis/ocean/climatology_map_argo.py]: @vanroekel, one more small thing.  This link name should be different for SOSE and ARGO.  I clicked on the image for ARGO in my test and it went to SOSE instead.  Sorry I didn't notice this before.  (Same for salinity, of course).  Please add 'Argo' and the end here and 'Sose' in the corresponding place in the SOSE tasks.

**Comment by xylar** [comment]
Oh, dear.  The SOSE tasks are now broken.  Should be very easy to fix but maybe I'll come by and work with you directly.

File: mpas_analysis/ocean/climatology_map_sose.py [hunk 6]
```diff
@@ -112,7 +112,7 @@ def __init__(self, config, mpasClimatologyTask,
                 '{}/SOSE_2005-2010_monthly_pot_temp_6000.0x' \
                 '6000.0km_10.0km_Antarctic_stereo.nc'.format(
                         observationsDirectory)
-            refFieldName = 'theta'
+            refFieldName = 'thetaSOSE'
```
Comment (xylar) [inline mpas_analysis/ocean/climatology_map_sose.py]: We also need to rename the variable in the observational data set to match this name.

**Comment by xylar** [comment]
I'll test on Edison as soon as I can (but cscratch seems to be hanging at the moment).  Then, I'll merge because I think @milenaveneziani doesn't really have time to review this at the moment.

**Comment by vanroekel** [comment]
One other thought for a test. Do you have a sense what will happen if a user requests a depth below the max observation depth?  The Argo dataset only extends to 2000m

________________________________

**Comment by xylar** [comment]
@vanroekel, it searches for the "nearest" level so the user would (probably unknowingly) get Argo values at 2000m.  This isn't terribly easy to avoid unless you wanted to somehow add this to the subplot title.  I would think the best thing would be to add a comment in the config file saying that -2000 is the minimum depth.  I can do that.

**Comment by xylar** [comment]
My tests worked great! Here are the results:
http://portal.nersc.gov/project/acme/xylar/argo/20180209.GMPAS-IAF.T62_oEC60to30v3wLI.cori-knl.afterSalinityFix/

**Comment by xylar** [comment]
@vanroekel, as usual, I can't delete your branch but you probably will want to do so.

**Comment by xylar** [comment]
Oops, we didn't add the new tasks to the documentation.  Something to follow up on with a future (small) PR.

**Comment by vanroekel** [comment]
Will do.  Thanks @xylar for your invaluable help in getting this PR together.  I appreciate it.

**Comment by milenaveneziani** [comment]
very nice: thanks to both of you!

**Pull request merged**